pkg/render/node.go

```diff
@@ -610,6 +631,19 @@ func (c *nodeComponent) nodeVolumes() []v1.Volume {
 	fileOrCreate := v1.HostPathFileOrCreate
 	dirOrCreate := v1.HostPathDirectoryOrCreate
 
+	var felixVolume v1.VolumeSource
```

Comment:
I see this being code duplicated in multiple files. How about we define a function that creates a generic volume, like 

```
func CertificateVolume(certMgmt *CertificateManagement, secretName string) v1.VolumeSource {
var volume v1.VolumeSource
if cerMgmt != nil {
	volume = v1.VolumeSource{
		EmptyDir: &v1.EmptyDirVolumeSource{},
	}
} else {
	volume = v1.VolumeSource{
		Secret: &v1.SecretVolumeSource{
			SecretName: secretName
		},
	}
return volume
}


```